chart/compass/templates/rewrite-correlation-headers-filter.yaml

```diff
@@ -0,0 +1,38 @@
+apiVersion: networking.istio.io/v1alpha3
+kind: EnvoyFilter
+metadata:
+  name: {{ .Chart.Name }}-correlation-headers-rewrite
+  namespace: {{ .Values.global.istio.namespace }}
+spec:
+  workloadLabels:
+    app: {{ .Values.global.istio.ingressgateway.workloadLabel }}
+  filters:
+    - filterConfig:
+        inlineCode: |
+          function expectedCorrelationHeaders()
+            local headers = {}
+            {{ range .Values.global.istio.ingressgateway.correlationHeaderRewriteFilter.expectedHeaders }}headers['{{ . }}']=true
+            {{ end }}
+            return headers
+          end
+          function envoy_on_request(request_handle)
+            local expectedHeaders = expectedCorrelationHeaders()
+            local reqHeaders = request_handle:headers()
+            local newRequestIDHeaderValue=''
+            for key, value in pairs(reqHeaders) do
+              if (expectedHeaders[key] and newRequestIDHeaderValue=='')then
```

Comment:
we need the `and newRequestIDHeaderValue==''` instead of breaking the loop when we find the first occurrence because we cannot break the loop :) https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/lua_filter#pairs

Reply:
looks ok, just add this as a comment in the lua script maybe for further reference ?